ui/imports/shared/controls/TransactionAddress.qml

```diff
@@ -0,0 +1,134 @@
+    */
+    property string contactPubKey
+
+    /*!
+       \qmlproperty \l{StatusAssetSettings} TransactionAddress::asset
+       Property holds asset settings for contact icon.
+    */
+    property StatusAssetSettings asset: StatusAssetSettings {
+        width: 36
+        height: 36
+        color: Utils.colorForPubkey(root.contactPubKey)
+        name: isImage ? d.contactData.displayIcon : nameText.text
+        isImage: d.isContact && d.contactData.displayIcon.includes("data")
+        isLetterIdenticon: d.isContact && !isImage
+        charactersLen: 2
+    }
+
+    implicitHeight: Math.max(identicon.height, contentColumn.height) + 12
+
+    QtObject {
+        id: d
+        readonly property var prefixAndAddress: Utils.splitToChainPrefixAndAddress(root.address)
+        readonly property bool isContactPubKeyValid: root.contactPubKey !== undefined && !!root.contactPubKey
+        readonly property var contactData: Utils.getContactDetailsAsJson(root.contactPubKey)
+        readonly property bool isContact: isContactPubKeyValid && contactData.isContact
+    }
+
+    RowLayout {
+        id: contentRow
+        anchors.verticalCenter: parent.verticalCenter
+        anchors.left: parent.left
+        anchors.right: parent.right
+
+
+        StatusSmartIdenticon {
+            id: identicon
+            Layout.alignment: Qt.AlignTop
+            asset: root.asset
+            name: nameText.text
+            ringSettings {
+                ringSpecModel: d.isContact ? Utils.getColorHashAsJson(d.contactData.publicKey) : []
+                ringPxSize: asset.width / 24
+            }
+            visible: d.isContact
+        }
+
+        ColumnLayout {
+            id: contentColumn
+            Layout.fillWidth: true
+            spacing: 0
+
+            StatusBaseText {
+                id: nameText
+                Layout.fillWidth: true
+                font.pixelSize: 15
+                color: Theme.palette.directColor1
+                text: {
+                    let name = ""
+                    if (d.isContact) {
+                        name = ProfileUtils.displayName(d.contactData.localNickname, d.contactData.name, d.contactData.displayName, d.contactData.alias)
+                    }
+                    return name || root.addressName
+                }
+                visible: !!text
+                elide: Text.ElideRight
+            }
+            StatusBaseText {
+                Layout.fillWidth: true
+                font.pixelSize: 15
+                color: Theme.palette.directColor1
+                wrapMode: Text.WrapAnywhere
+                text: {
+                    if(!!root.address == false)
+                        return ""
+                    if (d.prefixAndAddress.prefix.length > 0) {
+                        return WalletUtils.colorizedChainPrefix(d.prefixAndAddress.prefix) + d.prefixAndAddress.address  + "asdasdaspdjsajdasdasjpasdjaspdas"
```

Comment:
"asdasdaspdjsajdasdasjpasdjaspdas" leftover perhaps?

Reply:
Sorry, you are right, fixed.